pkg/util/optional.go

```diff
@@ -0,0 +1,26 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package util
+
+// Optional reprensents an optional type.
+// By default, no value is set and a call to GetValue() returns (T{}, false)
+type Optional[T any] struct {
+	value T
+	set   bool
+}
+
+// NewOptional creates a new Optional[T] type with a value. A call to GetValue() returns (value, true)
+func NewOptional[T any](value T) Optional[T] {
```

Comment:
Should we also provide a `Set` and possibly `UnSet` methods if we want to change the optional value after initialization?